src/hotspot/share/opto/addnode.cpp

```diff
@@ -91,9 +91,17 @@ static bool commute(Node *add, bool con_left, bool con_right) {
 
   PhiNode *phi;
   // Check for tight loop increments: Loop-phi of Add of loop-phi
-  if( in1->is_Phi() && (phi = in1->as_Phi()) && !phi->is_copy() && phi->region()->is_Loop() && phi->in(2)==add)
+#ifdef ASSERT
+  if (in1->is_Phi())
+    assert(!in1->as_Phi()->is_copy(), "in1 cannot be a copy");
```

Comment:
The if condition should be added to the assert (the `#ifdef ASSERT` can then be removed):
`assert(!in1->is_Phi() || !in1->as_Phi()->is_copy() ...`